### PR TITLE
Matpower. DC lines import

### PR DESCRIPTION
**Pull request by geofjamg**
Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@gmail.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
We do not take into account HVDC lines.
https://matpower.org/docs/MATPOWER-manual-5.0.pdf

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by geofjamg** [comment]
@annetill help welcome!

File: matpower/matpower-model/src/main/java/com/powsybl/matpower/model/MDcLine.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
```
Comment (zamarrenolm) [inline matpower/matpower-model/src/main/java/com/powsybl/matpower/model/MDcLine.java:2]: use an updated license header, must include SPDX reference

File: matpower/matpower-model/src/main/java/com/powsybl/matpower/model/MDcLine.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.matpower.model;
+
+/**
+ *
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
```
Comment (zamarrenolm) [inline matpower/matpower-model/src/main/java/com/powsybl/matpower/model/MDcLine.java:11]: fix the author

File: matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java [hunk 3]
```diff
@@ -433,6 +440,63 @@ private static double admittanceEndToEngineeringUnitsForLine(double transmission
...
+                    .setConnectableBus(bus2Id)
+                    .setVoltageRegulatorOn(true)
+                    .setVoltageSetpoint(mDcLine.getVt() * voltageLevel2.getNominalV())
+                    .setLossFactor((float) computeLossFactor2(mDcLine.getPf(), losses * 0.5, losses * 0.5))
+                    .add();
+            network.newHvdcLine()
+                    .setId(id)
+                    .setConverterStationId1(csId1)
+                    .setConverterStationId2(csId2)
+                    .setR(0)
```
Comment (geofjamg) [inline matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java:478]: There is no such information in the Matpower model?

**Review by geofjamg — COMMENTED** [review]
Do we add the export in this PR (could be a way to validate everything with a roundtrip) or in another PR?

File: matpower/matpower-model/src/main/java/com/powsybl/matpower/model/MDcLine.java (hunk 1 above)
Comment (marqueslanauja) [reply matpower/matpower-model/src/main/java/com/powsybl/matpower/model/MDcLine.java]: Done

File: matpower/matpower-model/src/main/java/com/powsybl/matpower/model/MDcLine.java (hunk 2 above)
Comment (marqueslanauja) [reply matpower/matpower-model/src/main/java/com/powsybl/matpower/model/MDcLine.java]: Done

File: matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java (hunk 3 above)
Comment (marqueslanauja) [reply matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java:478]: No, in matpower DC lines are modeled as two equivalent dummy generators.

image

**Comment by marqueslanauja** [comment]
> Do we add the export in this PR (could be a way to validate everything with a roundtrip) or in another PR?

We planned to do it in another PR.

**Comment by geofjamg** [comment]
@marqueslanauja @zamarrenolm PR is OK for me.

**Review by zamarrenolm — APPROVED** [review]

**Pull request merged**